Reimplement 2 tests using fakeexec

bart0sh opened the pull request:
**What this PR does / why we need it**:

Used fakeexec API from utils/exec/testing to test
GetKubeletVersion and KubeletVersionCheck APIs without running kubelet.

```release-note
NONE
```

kad commented:
/ok-to-test

kad commented on cmd/kubeadm/app/preflight/utils.go:
```diff
@@ -18,20 +18,21 @@ package preflight
+	out, err := command.Output()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to run kubelet: %v", err)
```
it will be good also to update tests to use fakeexec for GetKubeletVersion

bart0sh committed:
Reimplement 2 tests using fakeexec

Used fakeexec API from utils/exec/testing to test
GetKubeletVersion and KubeletVersionCheck APIs without
running kubelet.

kad commented:
Thanks @bart0sh for fixing that. It was in my todo list for quite a while.
/lgtm

bart0sh commented:
/retest

bart0sh commented:
@kad Can you /lgtm again, please? I've just rebased my branch in a hope it'd fix tests and didn't do any other changes.

kad commented:
/lgtm

bart0sh commented:
/area kubeadm

timothysc reviewed (approved):
/approve
